sdk/storage/inc/datalake/file_system_client.hpp

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "common/storage_credential.hpp"
+#include "common/storage_url_builder.hpp"
+#include "datalake/service_client.hpp"
+#include "datalake_options.hpp"
+#include "http/pipeline.hpp"
+#include "protocol/datalake_rest_client.hpp"
+
+#include <memory>
+#include <string>
+
+namespace Azure { namespace Storage { namespace DataLake {
+
+  class PathClient;
+
+  struct FileSystemGetMetadataResponse
+  {
+    std::string Date;
+    std::string ETag;
+    std::string LastModified;
+    std::string RequestId;
+    std::string Version;
+    std::map<std::string, std::string> Metadata;
+    std::string NamespaceEnabled;
+  };
+
+  typedef FileSystemSetPropertiesResponse FileSystemSetMetadataResponse;
```

Comment:
I prefer

```
using FileSystemSetMetadataResponse = FileSystemSetPropertiesResponse;
```

but i'm also fine with typedef.

Reply:
Can you elaborate on why using is preferred?

Reply:
nothing, just personal preference. Let's forget about it.